### PR TITLE
add LCM example

### PR DESCRIPTION
**Pull request by doombeaker**
RTX 3090, fp16, steps=4, warmup=5:

## pytorch

```bash
python diffusers/examples/text_to_image_LCM.py --model_id /share_nfs/hf_models/LCM_Dreamshaper_v7 --warmup 5 --disable
```

```bash
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00,  6.43it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 14.94it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 14.82it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 14.88it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 14.90it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 14.83it/s]
```

image

## onediff

```bash
python diffusers/examples/text_to_image_LCM.py --model_id /share_nfs/hf_models/LCM_Dreamshaper_v7 --warmup 5 
```

```bash
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:25<00:00,  6.35s/it]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 28.51it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 28.72it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 28.35it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 28.60it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 28.50it/s]
```

image

## conclusion

Roughly an acceleration of 90%

### CODE REVIEW

File: examples/requirements.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # pip install -r requirements.txt
 oneflow --pre -f https://staging.oneflow.info/branch/master/cu118
 transformers==4.27.1
-diffusers[torch]==0.19.3
+diffusers[torch]==0.23.0
```
Comment (doombeaker) [inline examples/requirements.txt:4]: LCM 要求最低是这个版本。https://huggingface.co/SimianLuo/LCM_Dreamshaper_v7

File: examples/text_to_image_LCM.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+if not args.disable:
+    from onediff.infer_compiler import oneflow_compile
+import torch
+
+pipe = DiffusionPipeline.from_pretrained(args.model_id)
+
+pipe.to(torch_device="cuda", torch_dtype=torch.float16)
+if not args.disable:
+    pipe.unet = oneflow_compile(pipe.unet)
```
Comment (strint) [inline examples/text_to_image_LCM.py:40]: RTX 3090, fp16, steps=4：
- origin:  28.5 it/s
- with oneflow_compile in onediff: 14.8 it/s

**Comment by doombeaker** [comment]
A100, fp16, steps=4

## pytorch

```
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00,  6.66it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 24.81it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 24.94it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 24.88it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 24.84it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 24.86it/s]
```

## onediff


```bash
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:24<00:00,  6.20s/it]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 44.37it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 44.06it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 43.76it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 42.00it/s]
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:00<00:00, 41.93it/s]
```

## conclusion

Roughly an acceleration of 73%

File: examples/requirements.txt (hunk 1 above)
Comment (strint) [reply examples/requirements.txt:4]: 这里是不是可以不改，另外专门写个 lcm 的 requirements.txt

然后在 lcm 例子里面检查下 diffusers 的版本，如果不是则提示用户用 lcm 专用的 requirements.txt

Comment (strint) [reply examples/requirements.txt:4]: 这样解决这块的合并问题

File: examples/text_to_image_LCM.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+for _ in range(args.warmup):
```
Comment (strint) [inline examples/text_to_image_LCM.py:42]: 可以给每个段落简单注释下，比如这段是 warmup 用的

File: examples/text_to_image_LCM.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import argparse
```
Comment (strint) [inline examples/text_to_image_LCM.py:1]: examples/text_to_image_LCM.py  -> examples/text_to_image_lcm.py

File: examples/requirements.txt (hunk 1 above)
Comment (doombeaker) [reply examples/requirements.txt:4]: 这如果没有其它特殊原因，是不是可以借这个机会把依赖升级了。

如果有2个依赖配置，对用户和 CI，都比较复杂和奇怪吧。

Comment (strint) [reply examples/requirements.txt:4]: > 这如果没有其它特殊原因，是不是可以借这个机会把依赖升级了。

升级 diffusers 的话那需要做一般系统的测试：
- 所有模型兼容性
- 正确性
- 速度
- 显存

到时我们估计发个新版本比较合适。

Comment (strint) [reply examples/requirements.txt:4]: 这样吧，直接在  lcm 例子里面检查下 diffusers 的版本，如果不是则提示用户安装新的 diffusers 版本就好。也不用动 examples/requirements.txt 了

Comment (doombeaker) [reply examples/requirements.txt:4]: > 升级 diffusers 的话那需要做一版系统的测试：

> 所有模型兼容性
> 正确性
> 速度
> 显存
> 动静有点大，到时我们估计发个新版本比较合适。

这些东西我们其实以前和现在都没系统测过，（目前为止）都是用户或者我们自己使用过程中反馈的问题。

我不是说不可以改两套 requriements.txt 的方式，也不是不认同上面的理由。

只是说，我们如果以却反系统测试而不升级依赖的话，估计未来很长一段时间，都不会升级了。

Comment (doombeaker) [reply examples/requirements.txt:4]: > 这样吧，直接在 lcm 例子里面检查下 diffusers 的版本，如果不是则提示用户安装新的 diffusers 版本就好。也不用动 examples/requirements.txt 了

这样我还是觉得会比较奇怪的，想象下：
用户按照我们的 README 安装了 onediff，然后试跑 lcm，然后被警告要升级 diffusers，但实际上它刚刚安装了仓库的 requirements.txt

Comment (strint) [reply examples/requirements.txt:4]: > 估计未来很长一段时间，都不会升级了。

这个对为了的3个月来说，按说也不太重要。

当前我们发出来的0.19.3，稳定性和质量是我们过去两个月内部不断试验各种情况，和最近社区一周的 issue 反馈得来的。

在没有什么保障的情况下，就升级一个版本，对内外部的开发节奏都可能带来一周的动荡。而升级的必要性又还不够强烈。

File: examples/text_to_image_LCM.py (hunk 3 above)
Comment (doombeaker) [reply examples/text_to_image_LCM.py:42]: 我在 warmup 的参数里加个说明吧

Comment (doombeaker) [reply examples/text_to_image_LCM.py]: 已经添加

File: examples/requirements.txt (hunk 1 above)
Comment (doombeaker) [reply examples/requirements.txt:4]: 好的，那我先按照针对跑 lcm 做版本检查的方式来吧

Comment (doombeaker) [reply examples/requirements.txt:4]: 已经改了

File: examples/text_to_image_LCM.py (hunk 4 above)
Comment (doombeaker) [reply examples/text_to_image_LCM.py]: 已改

File: examples/requirements.txt (hunk 1 above)
Comment (strint) [inline examples/requirements.txt:4]: 这个改动需要去掉

Comment (doombeaker) [reply examples/requirements.txt]: done

**Review by strint — APPROVED** [review]
